src/core/server/http/router/request.ts

```diff
@@ -73,23 +75,24 @@ export class KibanaRequest<Params = unknown, Query = unknown, Body = unknown> {
   public readonly path: string;
   public readonly url: Url;
 
+  protected readonly [requestSymbol]: Request;
+
   constructor(
-    private readonly request: Request,
+    request: Request,
     readonly params: Params,
     readonly query: Query,
     readonly body: Body
   ) {
     this.headers = request.headers;
     this.path = request.path;
     this.url = request.url;
+
+    this[requestSymbol] = request;
   }
 
   public getFilteredHeaders(headersToKeep: string[]) {
     return filterHeaders(this.headers, headersToKeep);
   }
-
-  // eslint-disable-next-line @typescript-eslint/camelcase
-  public unstable_getIncomingMessage() {
-    return this.request.raw.req;
-  }
 }
+
+export const toRawRequest = (request: KibanaRequest) => request[requestSymbol];
```

Comment:
this api shouldn't be exposed outside of `http service`

Reply:
suggestion: Because this is kind of sensitive, perhaps we should add a comment and mark it with `@internal`, it's easy to do a refactor like `export * from './request'` and accidentally expose this

Reply:
done